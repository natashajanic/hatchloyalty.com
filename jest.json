{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
