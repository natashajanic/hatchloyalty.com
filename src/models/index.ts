import { FluidObject } from 'gatsby-image'

export interface IBlogPost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    author: string
    date: string
    featuredImage?: IGraphQLImage
    tags: string[]
    title: string
  }
  html: string
}

export interface IGraphQLImage {
  childImageSharp: {
    fluid: FluidObject
  }
}

export interface IRelease {
  changes: Array<{ component: string; description: string; link?: string }>
  date: string
  name: string
}

export interface IResourcePost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    title: string
  }
  html: string
}

export interface ITeamMember {
  headshot?: IGraphQLImage
  name: string
  position: string
}
