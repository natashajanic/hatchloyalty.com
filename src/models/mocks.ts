import { IRelease, IBlogPost, IResourcePost, ITeamMember } from '.'

export const mockBlogPost: IBlogPost = {
  excerpt: 'This is the excerpt for a blog post.',
  fields: {
    slug: '/2019-02-25-test-blog-post/',
  },
  frontmatter: {
    author: 'John Doe',
    date: 'February 25, 2019',
    featuredImage: {
      childImageSharp: {
        fluid: {
          aspectRatio: 1,
          src: 'test/imgs/test-blog-post.png',
          srcSet: '',
          sizes: '',
        },
      },
    },
    tags: ['Test', 'Press Release'],
    title: 'Test Blog Post',
  },
  html: '<div>Test blog post body content.</div>',
}

export const mockRelease: IRelease = {
  changes: [
    { component: 'Test', description: 'Test Change 1' },
    {
      component: 'Test HQ',
      description: 'Test Change 2',
      link: 'https://developer.hatchloyalty.com/test/change-2',
    },
  ],
  date: 'January 2019',
  name: 'Test Release 2019.01',
}

export const mockResourcePost: IResourcePost = {
  excerpt: 'This is the excerpt for a resource post.',
  fields: {
    slug: '/test-resource-post/',
  },
  frontmatter: {
    title: 'Test Resource Post',
  },
  html: '<div>Test resource post body content.</div>',
}

export const mockTeamMember: ITeamMember = {
  headshot: {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        src: 'test/imgs/john-doe.png',
        srcSet: '',
        sizes: '',
      },
    },
  },
  name: 'John Doe',
  position: 'CEO',
}
