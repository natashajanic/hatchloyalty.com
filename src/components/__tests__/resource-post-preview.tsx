import { shallow, ShallowWrapper } from 'enzyme'
import { Link } from 'gatsby'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Text from 'src/components/text'
import { mockResourcePost } from 'src/models/mocks'
import ResourcePostPreview, {
  IResourcePostPreviewProps,
} from '../resource-post-preview'

describe('ResourcePostPreview', () => {
  let props: IResourcePostPreviewProps
  let mountedComponent: ShallowWrapper<any, any, ResourcePostPreview>

  beforeEach(() => {
    props = {
      post: mockResourcePost,
    }
    mountedComponent = shallow(<ResourcePostPreview {...props} />)
  })

  describe('render', () => {
    it('renders the resource title as a link to the post', async () => {
      mountedComponent.setProps({
        post: {
          ...mockResourcePost,
          fields: { ...mockResourcePost.fields, slug: '/test-resource-1/' },
          frontmatter: {
            ...mockResourcePost.frontmatter,
            title: 'Test Resource Post 1',
          },
        },
      })
      expect(mountedComponent.find(Link).length).toEqual(2)
      const titleLink = mountedComponent.find(Link).at(0)
      expect(titleLink.text()).toEqual('Test Resource Post 1')
      expect(titleLink.prop('to')).toEqual('/resource/test-resource-1/')
    })

    it("renders the post's excerpt", async () => {
      expect(mountedComponent.find(Text).length).toEqual(1)
      expect(
        mountedComponent
          .find(Text)
          .render()
          .text()
      ).toEqual(mockResourcePost.excerpt)
    })

    it('renders an explicit link to the blog post', async () => {
      mountedComponent.setProps({
        post: {
          ...mockResourcePost,
          fields: { ...mockResourcePost.fields, slug: '/test-resource-1/' },
        },
      })
      expect(mountedComponent.find(Link).length).toEqual(2)
      const readMoreLink = mountedComponent.find(Link).last()
      expect(readMoreLink.text()).toEqual('Read More')
      expect(readMoreLink.prop('to')).toEqual(`/resource/test-resource-1/`)
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
