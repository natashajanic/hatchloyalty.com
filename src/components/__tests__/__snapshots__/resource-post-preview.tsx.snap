// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourcePostPreview render matches the expected snapshot 1`] = `
<div>
  <ForwardRef
    style={
      Object {
        "color": "#666666",
        "display": "block",
        "textDecoration": "none",
      }
    }
    to="/resource/test-resource-post/"
  >
    Test Resource Post
  </ForwardRef>
  <Text
    blacklist={
      Array [
        "css",
        "fontSize",
        "lineHeight",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "textStyle",
        "alignItems",
        "color",
        "bg",
        "display",
        "flex",
        "fontWeight",
        "textAlign",
      ]
    }
    color="grayDark"
    dangerouslySetInnerHTML={
      Object {
        "__html": "This is the excerpt for a resource post.",
      }
    }
    fontSize={3}
    is="p"
    lineHeight="1.5"
    m={0}
    textStyle="normal"
  />
  <ForwardRef
    color="grayDark"
    to="/resource/test-resource-post/"
  >
    Read More
  </ForwardRef>
</div>
`;
